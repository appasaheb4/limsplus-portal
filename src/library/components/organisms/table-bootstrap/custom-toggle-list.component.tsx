import React, { useState, useEffect } from 'react';
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';

interface ColumnFilterProps {
  columns: Array<{ dataField: string; text: string }>;
  onClose: () => void;
  onColumnReorder: (
    newColumns: Array<{ dataField: string; text: string }>,
  ) => void;
  onColumnToggle: (selectedColumns: Array<string>) => void;
}

const ColumnFilter: React.FC<ColumnFilterProps> = ({
  columns,
  onClose,
  onColumnReorder,
  onColumnToggle,
}) => {
  const [selectedColumns, setSelectedColumns] = useState<string[]>(
    columns.map(column => column.dataField),
  );
  const [columnOrder, setColumnOrder] = useState(columns);
  const [searchQuery, setSearchQuery] = useState('');

  useEffect(() => {
    onColumnToggle(selectedColumns);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [selectedColumns]);

  useEffect(() => {
    onColumnReorder(columnOrder);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [columnOrder]);

  const handleToggle = (column: string) => {
    setSelectedColumns(prev =>
      prev.includes(column)
        ? prev.filter(c => c !== column)
        : [...prev, column],
    );
  };

  const handleSelectAll = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (event.target.checked) {
      setSelectedColumns(filteredColumns.map(column => column.dataField));
    } else {
      setSelectedColumns([]);
    }
  };

  const onDragEnd = result => {
    if (!result.destination) return;
    const reorderedColumns = Array.from(columnOrder);
    const [removed] = reorderedColumns.splice(result.source.index, 1);
    reorderedColumns.splice(result.destination.index, 0, removed);
    setColumnOrder(reorderedColumns);
  };

  const filteredColumns = columnOrder.filter(
    column =>
      column.dataField !== '_id' &&
      column.text.toLowerCase().includes(searchQuery.toLowerCase()),
  );

  const allSelected =
    filteredColumns.length > 0 &&
    filteredColumns.every(column => selectedColumns.includes(column.dataField));

  return (
    <div
      className='bg-white border rounded p-4 shadow-md absolute z-50'
      style={{ width: '300px', maxHeight: '400px', overflowY: 'auto' }}
    >
      <input
        type='text'
        placeholder='Search columns...'
        value={searchQuery}
        onChange={e => setSearchQuery(e.target.value)}
        className='mb-4 p-2 border rounded w-full'
      />
      <div className='mb-4'>
        <label className='flex items-center'>
          <input
            type='checkbox'
            checked={allSelected}
            onChange={handleSelectAll}
            className='form-checkbox'
          />
          <span className='ml-2'>Select All</span>
        </label>
      </div>
      <DragDropContext onDragEnd={onDragEnd}>
        <Droppable droppableId='columns'>
          {provided => (
            <div ref={provided.innerRef} {...provided.droppableProps}>
              {filteredColumns.map((column, index) => (
                <Draggable
                  key={column.dataField}
                  draggableId={column.dataField}
                  index={index}
                >
                  {provided => (
                    <div
                      ref={provided.innerRef}
                      {...provided.draggableProps}
                      {...provided.dragHandleProps}
                      className='mb-2'
                    >
                      <label className='flex items-center'>
                        <input
                          type='checkbox'
                          checked={selectedColumns.includes(column.dataField)}
                          onChange={() => handleToggle(column.dataField)}
                          className='form-checkbox'
                        />
                        <span className='ml-2'>{column.text}</span>
                      </label>
                    </div>
                  )}
                </Draggable>
              ))}
              {provided.placeholder}
            </div>
          )}
        </Droppable>
      </DragDropContext>
      <button
        onClick={onClose}
        className='mt-4 bg-blue-500 text-white py-2 px-4 rounded hover:bg-blue-700'
      >
        Close
      </button>
    </div>
  );
};

export default ColumnFilter;
