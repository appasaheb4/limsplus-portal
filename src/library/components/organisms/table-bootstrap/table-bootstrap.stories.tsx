/* eslint-disable react/jsx-indent-props */
/* eslint-disable no-console */
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { TableBootstrap } from './table-bootstrap.component';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Organisms/TableBootstrap',
  component: TableBootstrap,
} as ComponentMeta<typeof TableBootstrap>;

const columns = [
  {
    dataField: 'id',
    text: 'Product ID',
  },
  {
    dataField: 'name',
    text: 'Product Name',
    formatter: (cellContent, row) => {
      const maxLength = 5;
      const displayTestName =
        row.name.length > maxLength
          ? row.name.slice(0, Math.max(0, maxLength)) + '...'
          : row.name;
      return (
        <div className='flex flex-row'>
          <span title={row.name}>{`${displayTestName}`}</span>
        </div>
      );
    },
  },
  {
    dataField: 'price',
    text: 'Product Price',
  },
];
const data = [
  {
    id: '001',
    name: 'Limsplus',
    price: '100000',
  },
  {
    id: '001',
    name: 'Limsplus',
    price: '100000',
  },
  {
    id: '001',
    name: 'Limsplus',
    price: '100000',
  },
  {
    id: '001',
    name: 'Limsplus',
    price: '100000',
  },
  {
    id: '001',
    name: 'Limsplus',
    price: '100000',
  },
  {
    id: '001',
    name: 'Limsplus',
    price: '100000',
  },
];
export const _TableBootstrap: ComponentStory<typeof TableBootstrap> = () => (
  <TableBootstrap
    id='_id'
    columns={columns}
    data={data}
    fileName='StoryBook'
    totalSize={26}
    onDelete={() => {}}
    onFilter={() => {}}
    onSelectedRow={() => {}}
    onPageSizeChange={() => {}}
    onUpdateItem={() => {}}
    clearAllFilter={() => {}}
  />
);
