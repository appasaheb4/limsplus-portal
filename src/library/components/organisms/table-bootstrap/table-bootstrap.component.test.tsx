import React from 'react';
import {render} from '@testing-library/react';
import {TableBootstrap} from './table-bootstrap.component';

it('render autoComplete correctly', () => {
  const autoComplete = render(
    <TableBootstrap
      id='_id'
      data={[]}
      columns={[]}
      fileName='test'
      onSelectedRow={() => jest.fn()}
    />,
  );
  expect(autoComplete).toMatchSnapshot();
});
