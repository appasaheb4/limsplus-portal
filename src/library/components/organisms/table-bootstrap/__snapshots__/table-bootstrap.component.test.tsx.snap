// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBootstrap component render tableBootstrap correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex items-center"
        >
          <label
            class="search-label"
            for="search-bar-0"
          >
            <span
              class="sr-only"
              id="search-bar-0-label"
            >
              Search this table
            </span>
            <input
              aria-labelledby="search-bar-0-label"
              class="form-control "
              id="search-bar-0"
              placeholder="Search..."
              type="text"
              value=""
            />
          </label>
          <button
            class="btn btn-default inline-flex ml-4 bg-gray-500 items-center small outline shadow-sm  font-medium  disabled:opacity-50 disabled:cursor-not-allowed text-center h-9 text-white"
          >
            Clear
          </button>
          <button
            class="ml-2 px-2 focus:outline-none bg-gray-500 items-center  outline shadow-sm  font-medium  text-center rounded-md h-9 text-white"
          >
            Clear all filters
          </button>
          <button
            class="react-bs-table-csv-btn btn btn-default inline-flex m-2.5 bg-gray-500 items-center  small outline shadow-sm  font-medium  disabled:opacity-50 disabled:cursor-not-allowed text-center h-9 text-white"
            type="button"
          >
            Export CSV!!
          </button>
          <div
            class="undefined"
          >
            <button
              class="undefined inline-flex items-center px-3 py-2 text-sm rounded-lg shadow-sm   font-medium text-gray-600 border border-gray-400 hover:shadow-lg disabled:opacity-50 disabled:cursor-not-allowed text-center"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="scrollTable"
        >
          <div
            class="react-bootstrap-table"
          >
            <table
              class="table table-hover table-bordered"
            >
              <thead>
                <tr
                  class="bg-gray-500 text-white whitespace-nowrap"
                >
                  <th
                    tabindex="0"
                  >
                    Product ID
                  </th>
                  <th
                    tabindex="0"
                  >
                    Product Name
                  </th>
                  <th
                    tabindex="0"
                  >
                    Product Price
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    001
                  </td>
                  <td>
                    Limsplus
                  </td>
                  <td>
                    100000
                  </td>
                </tr>
                <tr>
                  <td>
                    001
                  </td>
                  <td>
                    Limsplus
                  </td>
                  <td>
                    100000
                  </td>
                </tr>
                <tr>
                  <td>
                    001
                  </td>
                  <td>
                    Limsplus
                  </td>
                  <td>
                    100000
                  </td>
                </tr>
                <tr>
                  <td>
                    001
                  </td>
                  <td>
                    Limsplus
                  </td>
                  <td>
                    100000
                  </td>
                </tr>
                <tr>
                  <td>
                    001
                  </td>
                  <td>
                    Limsplus
                  </td>
                  <td>
                    100000
                  </td>
                </tr>
                <tr>
                  <td>
                    001
                  </td>
                  <td>
                    Limsplus
                  </td>
                  <td>
                    100000
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="row react-bootstrap-table-pagination"
          >
            <div
              class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
            />
            <div
              class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
            >
              <ul
                class="pagination react-bootstrap-table-page-btns-ul"
              >
                <li
                  class="disabled page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                  >
                    &lt;&lt;
                  </a>
                </li>
                <li
                  class="disabled page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="active page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                  >
                    0
                  </a>
                </li>
                <li
                  class="page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                  >
                    1
                  </a>
                </li>
                <li
                  class="page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                  >
                    2
                  </a>
                </li>
                <li
                  class="page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                  >
                    &gt;
                  </a>
                </li>
                <li
                  class="page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                  >
                    &gt;&gt;
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="flex items-center gap-2 mt-2"
        >
          <div
            class="btn-group items-center"
            role="group"
          >
            <input
              class="mr-2 ml-2 leading-4 p-2 w-14 focus:outline-none focus:ring block  shadow-sm sm:text-base border border-gray-300 rounded-md"
              min="0"
              placeholder="No"
              type="number"
            />
            <button
              class="btn  bg-primary"
              type="button"
            >
              10
            </button>
            <button
              class="btn  bg-grey"
              type="button"
            >
              20
            </button>
            <button
              class="btn  bg-grey"
              type="button"
            >
              30
            </button>
            <button
              class="btn  bg-grey"
              type="button"
            >
              40
            </button>
            <button
              class="btn  bg-grey"
              type="button"
            >
              50
            </button>
          </div>
        </div>
        <div
          class="flex items-center gap-2 mt-2"
        >
          <div
            class="clearfix"
          />
          <span>
            Showing 
            1
             to 
            6
             of 
            6
             Results
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex items-center"
      >
        <label
          class="search-label"
          for="search-bar-0"
        >
          <span
            class="sr-only"
            id="search-bar-0-label"
          >
            Search this table
          </span>
          <input
            aria-labelledby="search-bar-0-label"
            class="form-control "
            id="search-bar-0"
            placeholder="Search..."
            type="text"
            value=""
          />
        </label>
        <button
          class="btn btn-default inline-flex ml-4 bg-gray-500 items-center small outline shadow-sm  font-medium  disabled:opacity-50 disabled:cursor-not-allowed text-center h-9 text-white"
        >
          Clear
        </button>
        <button
          class="ml-2 px-2 focus:outline-none bg-gray-500 items-center  outline shadow-sm  font-medium  text-center rounded-md h-9 text-white"
        >
          Clear all filters
        </button>
        <button
          class="react-bs-table-csv-btn btn btn-default inline-flex m-2.5 bg-gray-500 items-center  small outline shadow-sm  font-medium  disabled:opacity-50 disabled:cursor-not-allowed text-center h-9 text-white"
          type="button"
        >
          Export CSV!!
        </button>
        <div
          class="undefined"
        >
          <button
            class="undefined inline-flex items-center px-3 py-2 text-sm rounded-lg shadow-sm   font-medium text-gray-600 border border-gray-400 hover:shadow-lg disabled:opacity-50 disabled:cursor-not-allowed text-center"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="scrollTable"
      >
        <div
          class="react-bootstrap-table"
        >
          <table
            class="table table-hover table-bordered"
          >
            <thead>
              <tr
                class="bg-gray-500 text-white whitespace-nowrap"
              >
                <th
                  tabindex="0"
                >
                  Product ID
                </th>
                <th
                  tabindex="0"
                >
                  Product Name
                </th>
                <th
                  tabindex="0"
                >
                  Product Price
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  001
                </td>
                <td>
                  Limsplus
                </td>
                <td>
                  100000
                </td>
              </tr>
              <tr>
                <td>
                  001
                </td>
                <td>
                  Limsplus
                </td>
                <td>
                  100000
                </td>
              </tr>
              <tr>
                <td>
                  001
                </td>
                <td>
                  Limsplus
                </td>
                <td>
                  100000
                </td>
              </tr>
              <tr>
                <td>
                  001
                </td>
                <td>
                  Limsplus
                </td>
                <td>
                  100000
                </td>
              </tr>
              <tr>
                <td>
                  001
                </td>
                <td>
                  Limsplus
                </td>
                <td>
                  100000
                </td>
              </tr>
              <tr>
                <td>
                  001
                </td>
                <td>
                  Limsplus
                </td>
                <td>
                  100000
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="row react-bootstrap-table-pagination"
        >
          <div
            class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
          />
          <div
            class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
          >
            <ul
              class="pagination react-bootstrap-table-page-btns-ul"
            >
              <li
                class="disabled page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  &lt;&lt;
                </a>
              </li>
              <li
                class="disabled page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  &lt;
                </a>
              </li>
              <li
                class="active page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  0
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  1
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  2
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  &gt;
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href="#"
                >
                  &gt;&gt;
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-2 mt-2"
      >
        <div
          class="btn-group items-center"
          role="group"
        >
          <input
            class="mr-2 ml-2 leading-4 p-2 w-14 focus:outline-none focus:ring block  shadow-sm sm:text-base border border-gray-300 rounded-md"
            min="0"
            placeholder="No"
            type="number"
          />
          <button
            class="btn  bg-primary"
            type="button"
          >
            10
          </button>
          <button
            class="btn  bg-grey"
            type="button"
          >
            20
          </button>
          <button
            class="btn  bg-grey"
            type="button"
          >
            30
          </button>
          <button
            class="btn  bg-grey"
            type="button"
          >
            40
          </button>
          <button
            class="btn  bg-grey"
            type="button"
          >
            50
          </button>
        </div>
      </div>
      <div
        class="flex items-center gap-2 mt-2"
      >
        <div
          class="clearfix"
        />
        <span>
          Showing 
          1
           to 
          6
           of 
          6
           Results
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
